numba_rvsdg/tests/simulator.py

```diff
@@ -327,8 +330,11 @@ def op_FOR_ITER(self, inst):
         try:
             ind = next(tos)
         except StopIteration:
-            self.stack.pop()
             self.branch = True
+            if PYVERSION <= (3, 11):
```

Comment:
I think it would be best to use the explicit `if PYVERSION` construct as we do in Numba:

```python
if PYVERSION in ((3, 12),):
    ...
elif PYVERSION in ((3, 11),):
    ...
else:
    raise NotImplementedError(PYVERSION)
```

IIRC correctly `numba_rvsdg` was initially developed on 3.11 so I guess we only support that version and beyond.